### DISCUSSION TITLE
Revit database explorer (RDBE)

### DISCUSSION THREAD

**Discussion by NeVeSpl** [jeremytammik/RevitLookup#141] [Show and tell]
Let me join the race to build the first fully working WPF version of RevitLookup. I know that [Nice3point](https://github.com/Nice3point) is working on one #124 here and that [chuongmep](https://github.com/chuongmep) is working on [RevitLookupWpf](https://github.com/weianweigan/RevitLookupWpf). I started as the last one (two weeks ago), but mine is almost ready :blush:, there are still a few things to correct, but it is already usable:

[Revit database explorer (RDBE)](https://github.com/NeVeSpl/RevitDBExplorer)

![possibility-to-query-Revit-database-from-UI](https://raw.githubusercontent.com/NeVeSpl/RevitDBExplorer/main/documentation/examples/possibility-to-query-Revit-database-from-UI.gif)

**Comment by jeremytammik** [comment]
wow! an exciting race in the starting phase! it will be interesting to see how things pan out in the long haul. optimally, maybe the three separate projects can converge and exploit synchronicity... good luck and much success to all three!

**Comment by jeremytammik** [comment]
i love the single-line console interpreting string input to generate corresponding filtered element collectors!

**Comment by jeremytammik** [comment]
are you also aware of the initiative of @chuongmep  to [reinvigorate RevitPythonShell and hook it up with RevitLookup](https://thebuildingcoder.typepad.com/blog/2022/07/mep-api-intro-and-revitpythonshell-lookup-nuptials.html#3)? A connection like that would be a great fit for your command line interface as well.

**Comment by Nice3point** [comment]
Sounds interesting, my team and I will update the interface as soon as we have time, there will also be a search, OTA update and new context menu elements. Here it is important to understand that what is needed is not WPF / WinForms, but the possibility of extending the functionality. It’s more difficult to do this on Forms, where the logic is mixed with the UI, which is bad

**Comment by NeVeSpl** [comment]
I thought even about building in C# shell(scripting + cmd line) into RDBE. But apparently, there is no big demand for running C# code directly in Revit without using VS ([Revit.ScriptCS](https://github.com/sridharbaldava/Revit.ScriptCS) does not look like to be popular). 

In the end, I want to achieve "google search" like experience for querying through the line interface, I do not want a user to care or know much about FilteredElementCollector, and still able to find what he or she is looking for.

**Comment by NeVeSpl** [comment]
That were two very productive weeks for RDBE,  the final version is coming closer. RDBE learned how to deal with parameters in queries:
![possibility-to-query-Revit-database-from-UI](https://raw.githubusercontent.com/NeVeSpl/RevitDBExplorer/main/documentation/examples/rdq-revit-database-query-with-rql-revit-query-language.gif)
My favourite is `?!` operator, which allows very easily to find all elements that have a given parameter, regardless if it has value or not. 

![possibility-to-query-Revit-database-from-UI](https://raw.githubusercontent.com/NeVeSpl/RevitDBExplorer/main/documentation/examples/themes.gif)
Also, RDBE has got a new theme : "Dark - be ready"   😆 🤣

**Reply by chuongmep** [reply]
Awesome,  unfortunately my personal computer is no longer licensed to compete with you, this will help all the programmers who are working with revit ! 🥰
